modules/ghlint-snakeyaml/src/main/kotlin/net/twisterrob/ghlint/yaml/SnakeYaml.kt

```diff
@@ -1,54 +1,54 @@
 package net.twisterrob.ghlint.yaml
 
-import net.twisterrob.ghlint.analysis.Analyzer
+import net.twisterrob.ghlint.model.File
 import net.twisterrob.ghlint.model.RawFile
 import net.twisterrob.ghlint.model.SnakeComponentFactory
-import net.twisterrob.ghlint.model.Workflow
-import net.twisterrob.ghlint.results.Finding
-import net.twisterrob.ghlint.ruleset.RuleSet
+import net.twisterrob.ghlint.model.name
 import org.intellij.lang.annotations.Language
 import org.snakeyaml.engine.v2.api.Dump
 import org.snakeyaml.engine.v2.api.DumpSettings
-import org.snakeyaml.engine.v2.api.LoadSettings
 import org.snakeyaml.engine.v2.api.StreamDataWriter
-import org.snakeyaml.engine.v2.common.ScalarStyle
-import org.snakeyaml.engine.v2.composer.Composer
+import org.snakeyaml.engine.v2.nodes.MappingNode
 import org.snakeyaml.engine.v2.nodes.Node
-import org.snakeyaml.engine.v2.nodes.ScalarNode
-import org.snakeyaml.engine.v2.nodes.Tag
-import org.snakeyaml.engine.v2.parser.ParserImpl
-import org.snakeyaml.engine.v2.scanner.StreamReader
-import org.snakeyaml.engine.v2.schema.JsonSchema
 import java.io.StringWriter
-import kotlin.jvm.optionals.getOrElse
 
 public object SnakeYaml {
 
 	private val factory = SnakeComponentFactory()
 
-	public fun analyze(files: List<RawFile>, ruleSets: List<RuleSet>): List<Finding> {
-		val workflows = files.map(::loadWorkflow)
-		return Analyzer().analyzeWorkflows(workflows, ruleSets)
+	@Suppress("LiftReturnOrAssignment", "detekt.ReturnCount")
+	public fun load(file: RawFile): File {
+		val node = try {
+			factory.loadYaml(file)
+		} catch (@Suppress("detekt.TooGenericExceptionCaught") ex: Exception) {
+			return ErrorInvalidContent.create(file, ex).parent
+		}
+		if (node !is MappingNode) {
+			val error = IllegalArgumentException("Root node is not a mapping: ${node::class.java.simpleName}.")
+			return ErrorInvalidContent.create(file, error).parent
+		}
+		try {
+			return loadUnsafe(file, node)
+		} catch (@Suppress("detekt.TooGenericExceptionCaught") ex: Exception) {
+			return ErrorInvalidContent.create(file, ex).parent
+		}
```

Comment:
specific error type? + unknown error in syntax error -> syntaxerrorrule is not just syntax